commonv5/ts/helpers/__tests__/java-version-resolver-test.ts

```diff
@@ -1,61 +1,71 @@
-    const expectedJavaPath = "/opt/bin/java/bin";
-    const jdkSource = "JAVA_HOME";
-    jest.spyOn(tl, "getInput").mockReturnValueOnce(jdkSource);
-
-    jest.spyOn(tl, "getVariable").mockReturnValueOnce(expectedJavaPath); // JAVA_HOME
-
-    JavaVersionResolver.setJavaVersion(jdkSource);
-
-    const actualJavaHome = tl.getVariable(TaskVariables.JavaHome);
-
-    expect(actualJavaHome).toBe(expectedJavaPath);
-  });
-
-  it("JAVA_HOME_11_X64 set in task configuration, should switch to it", () => {
-    const java11Path = "/opt/bin/java11/bin";
-
-    const jdkSource = "JAVA_HOME_11_X64";
-    jest.spyOn(tl, "getInput").mockReturnValueOnce(jdkSource);
+const MOCKED_JAVA_VARIABLES = {
+  [JdkVersionSource.JavaHome]: "/opt/bin/java/bin",
+  [JdkVersionSource.JavaHome11]: "/opt/bin/java11/bin",
+  [JdkVersionSource.JavaHome17]: "/opt/bin/java17/bin",
+};
 
-    jest.spyOn(tl, "getVariable").mockReturnValueOnce(java11Path); // JAVA_HOME_11_X64
-
-    JavaVersionResolver.setJavaVersion(jdkSource);
-
-    const actualJavaHome = tl.getVariable(TaskVariables.JavaHome);
-
-    expect(actualJavaHome).toBe(java11Path);
-  });
+beforeEach(() => {
+  jest.spyOn(tl, "getVariable").mockImplementation((name) => MOCKED_JAVA_VARIABLES[name]);
+  jest.spyOn(tl, "setVariable");
+  jest.spyOn(tl, "warning");
 });
 
-describe("lookupVariable", () => {
-  it("java_home_11_x64 variable found, returning it", () => {
-    const javaHome11X64Value = "/opt/bin/java11/bin";
-    const jdkSource = "JAVA_HOME_11_X64";
+describe("JavaVersionResolver", () => {
+  it.each([
+    [EndpointType.SonarQube, "9.9.0"],
+    [EndpointType.SonarQube, "10.4.0"],
+    [EndpointType.SonarCloud, "8.0.0"],
+  ])("should not have an effect if chosing JAVA_HOME", (endpointType, serverVersion) => {
+    JavaVersionResolver.setJavaVersion(JdkVersionSource.JavaHome, endpointType, serverVersion);
 
-    jest.spyOn(tl, "getVariable").mockReturnValueOnce(javaHome11X64Value); // JAVA_HOME_11_X64
+    // Should not have changed the JAVA_HOME
+    expect(tl.setVariable).not.toHaveBeenCalled();
 
-    const actualJavaPath = JavaVersionResolver.lookupVariable(jdkSource);
-
-    expect(actualJavaPath).toBe(javaHome11X64Value);
+    // Revert action has no effect as well
+    JavaVersionResolver.revertJavaHomeToOriginal();
+    expect(tl.setVariable).not.toHaveBeenCalled();
   });
 
-  it("variable wanted not found, returning undefined", () => {
-    const jdkSource = "JAVA_HOME_11_X64";
-
-    jest.spyOn(tl, "getVariable").mockReturnValueOnce(undefined); // JAVA_HOME_11_X64
-
-    const actualJavaPath = JavaVersionResolver.lookupVariable(jdkSource);
-
-    expect(actualJavaPath).toBe(undefined);
-  });
+  it.each([
+    [EndpointType.SonarQube, undefined, JdkVersionSource.JavaHome11, "/opt/bin/java11/bin"],
+    [EndpointType.SonarQube, "9.9.0", JdkVersionSource.JavaHome11, "/opt/bin/java11/bin"],
+    [EndpointType.SonarQube, "9.9.0", JdkVersionSource.JavaHome17, "/opt/bin/java17/bin"],
```

Comment:
out of curiosity, why do we not test SQ 10.4 here?

Reply:
it's a good point, I'll add a test for SQ 10.4 + Java 17 to be more exhaustive